### ISSUE TITLE
Fix copying of ApplicationException popup messages

### ISSUE THREAD

**Issue by ghost** [octobercms/october#4704]
I think trying to create a piece of code in October is not as friendly as it could be. With the error messages being displayed using sweet alert vendor package. I can't copy the error message to do a web search to find the solution.

This got me thinking and wanted to open a github issue to try and make things a little bit better.

### Solution 1

- Allow the error message to be copy and paste.

### Solution 2

- Allow a button to be pressed that will search the internet for the displayed error message.

### Example

Before:

![image](https://user-images.githubusercontent.com/55155131/67152254-25065b80-f2ca-11e9-8345-cc1c27395150.png)

After:

![image](https://user-images.githubusercontent.com/55155131/67152258-2df72d00-f2ca-11e9-846c-d25bde987410.png)

In the case of the 'Search on Google' button the link would be something like this:

```html
<a class="btn btn-primary" href="https://www.google.com/search?q=array+to+string+conversion" rel="noopener noreferrer" target="_blank">Search on Google</a>
```

Then to make it a bit more customizable allow the user to select their own search link in the settings or config file. For example, someone may want to Search in Stackoverflow, or search in OctoberCMS website or a different search engine etc.

I leave it for people to make suggestions now.

**Comment by bennothommo** [comment]
@ayumihamasaki2019 I think it's a great idea, as long as it's only shown in debug mode.

**Comment by ayumihamasaki2019** [comment]
@bennothommo Thanks, yah good point "**only shown in debug mode**" makes total sense. As it should be shown only to developers and not clients (just expanding your comment if anyone else reads this issue).

Just wondering if October uses a custom file to generate the error messages or it's coming straight out of the javascript vendor file which would be a major issue.

**Comment by Rike-cz** [comment]
Sorry, not for. Most of errors needs to use brain instead of Google, some of them means something different than error message says. But it's only my opinion.

**Comment by bennothommo** [comment]
@ayumihamasaki2019 The popup is part of the Storm UI library. It's this file in particular that controls nearly all the popups: https://github.com/octobercms/october/blob/master/modules/system/assets/ui/js/popup.js

@Rike-cz I feel it's simply a shortcut to Google, and not necessarily something that is going to completely solve all issues, but it will help to encourage people to be a bit more proactive with errors and perhaps finding if other people have had the same issue.

**Comment by ayumihamasaki2019** [comment]
Yah, I do this method with many php and laravel error messages. I usually get my answer though from Stack Overflow. But I start with Google as they have the years listed next to the answers and I can choose a more modern answer then.

I can give a real life example below:

Error Message: **"The Response content must be a string or object implementing __toString(), "object" given." on line 409 of /home/example/public_html/vendor/symfony/http-foundation/Response.php**

The button link would then send me to the following web page:

![image](https://user-images.githubusercontent.com/55155131/67160366-e7e0ae80-f347-11e9-9057-b788d0bf0402.png)

Then I can research the search engine results to find a quick solution.

This method helps me to find quick answers on the fly and not needing to waste the admins time here posting a question.

Thanks bennothommo for the link, I couldn't find it when I did a search.

**Comment by LukeTowers** [comment]
@bennothommo @ayumihamasaki2019 while I think that the "Search on Google" UX is clever and cute, I don't feel like we should be encouraging that behaviour for every error. I think the proper solution to this (which would also solve other related problems) is just to fix that popup so that users can copy and paste the message like they would with anything else.

**Comment by Samuell1** [comment]
More like this is like "Warning that something went wrong" developers should look in error log where they can copy and see more about that error.

**Comment by ayumihamasaki2019** [comment]
@LukeTowers to expand on your comment:

My three main issues with the error message popup are as follows:

1. Can not copy and paste the error - have to go into the dom to copy it.

2. Add keyboard navigation to the error message popup - you can't close that modal using keyboard controls, you can only close it by mouse clicking. Haven't tested it with touch screen though.

3. Connection issues (e.g. Cloudflare Server Code) display a bunch of escaped HTML message, it would be nice if the HTML message was passed through something like HTMLPurifier and not show basic HTML escaped.

**Comment by drmzio** [comment]
Not a big fan of this. Maybe link to the log page instead? Also from a UX perspective, the blue button is **competing** with the OK button. Maybe make it a secondary button or just a link button with the arrow (↗️) that indicates an external link. Just my thoughts.

**Comment by LarBearrr** [comment]
Totally agree that copy and paste should be available in the popups. I think this issue may have been started because of the new Laravel 6 error page ([Ignition](https://laravel-news.com/ignition)), which implements all sorts of features to make it easier to find information on exceptions. @LukeTowers As October gets closer to upgrading to Laravel 6, do we know if we are going to support Ignition?

**Comment by LukeTowers** [comment]
@LarBearrr I'm certainly not opposed to it, haven't looked into adding it myself at all. If someone can get it working as a plugin I'm sure I would be willing to take the time to review it for merging.

**Comment by nizzac** [comment]
What's wrong with highting what you're wanting to copy and copying it as normal?

**Comment by LukeTowers** [comment]
@nizzac as I recall you can't actually do that on that popup for some reason. It's very annoying.

**Comment by bennothommo** [comment]
All, please feel free to test #4740 and let me know if this resolves the copy+paste issue for you.

**Comment by bennothommo** [comment]
Fixed by #4740.
